src/core/qgsvectorlayer.cpp

```diff
@@ -162,6 +162,8 @@ QgsVectorLayer::QgsVectorLayer( const QString &vectorLayerPath,
   mJoinBuffer->setParent( this );
   connect( mJoinBuffer, &QgsVectorLayerJoinBuffer::joinedFieldsChanged, this, &QgsVectorLayer::onJoinedFieldsChanged );
 
+  mServerProperties = new QgsVectorLayerServerProperties( this );
```

Comment:
Can we use a `unique_ptr` instead?

Reply:
I have done the same as `mJoinBuffer`. I don't know how to user `unique_ptr`

Reply:
You can find some examples of `unique_ptr` in `QgsProject` (`mLayerStore`, `mArchive`, `mLayoutManager`, ...).

Reply:
Done